src/SFA.DAS.EmployerCommitmentsV2.Web/Controllers/ApprenticeController.cs

```diff
@@ -467,6 +469,53 @@ public IActionResult HasTheApprenticeBeenMadeRedundant(MadeRedundantViewModel vi
         }
 
+        [Route("{apprenticeshipHashedId}/details/whystopapprenticeship")]
+        [HttpGet]
+        public async Task<IActionResult> WhyStopApprenticeship(WhyStopApprenticeshipRequest request)
+        {
+            var viewModel = await _modelMapper.Map<WhyStopApprenticeshipViewModel>(request);
+            return View(viewModel);
+        }
+
+        [Route("{apprenticeshipHashedId}/details/whystopapprenticeship",Name = RouteNames.WhyStopApprenticeship)]
+        [HttpPost]
+        public IActionResult WhyStopApprenticeship(WhyStopApprenticeshipViewModel viewModel)
+        {
+            switch (viewModel.SelectedStatusChange)
+            {
+                case StopStatusReason.LeftEmployment:
+                    return RedirectToAction(nameof(StopApprenticeship), new { viewModel.AccountHashedId, viewModel.ApprenticeshipHashedId });
+
+                case StopStatusReason.ChangeProvider:
+                    return RedirectToAction(nameof(StopApprenticeship), new { viewModel.AccountHashedId, viewModel.ApprenticeshipHashedId });
+
+                case StopStatusReason.Withdrawn:
+                    return RedirectToAction(nameof(StopApprenticeship), new { viewModel.AccountHashedId, viewModel.ApprenticeshipHashedId });
+
+                case StopStatusReason.NeverStarted:
+                    return RedirectToAction(nameof(ApprenticeshipNeverStarted), new { viewModel.AccountHashedId, viewModel.ApprenticeshipHashedId });
+
+                default:
+                    return RedirectToAction(nameof(ApprenticeshipDetails), new ApprenticeshipDetailsRequest { AccountHashedId = viewModel.AccountHashedId, ApprenticeshipHashedId = viewModel.ApprenticeshipHashedId });
+            }
+            //return RedirectToAction(nameof(HasTheApprenticeBeenMadeRedundant), new ApprenticeshipDetailsRequest { AccountHashedId = viewModel.AccountHashedId, ApprenticeshipHashedId = viewModel.ApprenticeshipHashedId });
+        }
+
+        [Route("{apprenticeshipHashedId}/details/apprenticeshipneverstarted")]
+        [HttpGet]
+        public async Task<IActionResult> ApprenticeshipNeverStarted(ApprenticeshipNeverStartedRequest request)
+        {
+            var viewModel = await _modelMapper.Map<ApprenticeshipNeverStartedViewModel>(request);
+            return View(viewModel);
+        }
+
+        [Route("{apprenticeshipHashedId}/details/apprenticeshipneverstarted", Name = RouteNames.ApprenticeshipNeverStarted)]
+        [HttpPost]
+        public IActionResult ApprenticeshipNeverStarted(ApprenticeshipNeverStartedViewModel viewModel)
+        {
+            return RedirectToAction(nameof(HasTheApprenticeBeenMadeRedundant), new ApprenticeshipDetailsRequest { AccountHashedId = viewModel.AccountHashedId, ApprenticeshipHashedId = viewModel.ApprenticeshipHashedId });

```

Comment:
This should pass an anonymous object as the model to the `RedirectToAction()` and the start date needs to be Posted to this action and then passed into that anonymous object the same way as is done on the normal stop journey: 
https://github.com/SkillsFundingAgency/das-employercommitments-v2/blob/e0423a174caa119331c3c0dd152f2ce5d96d580a/src/SFA.DAS.EmployerCommitmentsV2.Web/Controllers/ApprenticeController.cs#L451

Reply:
Added the extra parameters and updated the viewModel